filter.h

```diff
@@ -0,0 +1,176 @@
+#ifndef _FILTER_H_
+#define _FILTER_H_
+
+//Standard Includes 
+#include <stdint.h>
+
+
+
+/** @file
+ * @defgroup MovingAvg Filter Module   
+ * 	@ingroup Filters
```

Comment:
The group, Filters, does not exist that I can see. I would suggest just changing the @defgroup to 
@defgroup filter Filter Module
but if you want sub modules for the filter functions you would use the @defgroup and @ingroup commands like you do above but the filters group would need to be created/documented first.

Also, the rest of the documentation is not associated with any module. You can use @{ in a comment block and then @} in another comment block to group all comment blocks in between the two together (look at task.h for example).